dotnet/samples/KernelSyntaxExamples/Example76_Filters.cs

```diff
@@ -99,30 +197,132 @@ public FirstPromptFilter(ITestOutputHelper output)
 
     #region Filter capabilities
 
+    /// <summary>Shows syntax for function filter in non-streaming scenario.</summary>
     private sealed class FunctionFilterExample : IFunctionFilter
     {
-        public void OnFunctionInvoked(FunctionInvokedContext context)
+        public async Task OnFunctionInvocationAsync(FunctionInvocationContext context, Func<FunctionInvocationContext, Task> next)
         {
-            // Example: get function result value
-            var value = context.Result.GetValue<object>();
+            // Example: override kernel arguments
+            context.Arguments["input"] = "new input";
 
-            // Example: override function result value
-            context.SetResultValue("new result value");
+            // This call is required to proceed with next filters in pipeline and actual function.
+            // Without this call next filters and function won't be invoked.
+            await next(context);
+
+            // Example: get function result value
+            var value = context.Result!.GetValue<object>();
 
             // Example: get token usage from metadata
             var usage = context.Result.Metadata?["Usage"];
+
+            // Example: override function result value
+            context.Result = new FunctionResult(context.Function, "Result from filter");
```

Comment:
If we allows filters to modify the metadata then this will need to change.

Reply:
@markwallace-microsoft, @dmytrostruk If we want to change metadata wouldn't this be the way?

```suggestion
            var modifiedMetadata = new Dictionary<string, object?>(context.Result.Metadata)) 
            {
                ["new-data-index"] = "new-data-value"
            }
            context.Result = new FunctionResult(context.Function, "Result from filter", modifiedMetadata);
```

Reply:
@RogerBarreto Yes, this should be the pattern. I will add more tests and examples to make sure it is working as expected.

Reply:
I've added more tests and examples to verify that updating Metadata works as expected.